### ISSUE TITLE
Take out css definitions to separate lib

### ISSUE THREAD

**Issue by whitecolor** [typestyle/typestyle#200]
Typestyle has good css props typings with docs. This could be also used for some other libraries related to css.

One of the use cases: https://github.com/snabbdom/snabbdom/issues/339

What do you think about taking out css props definitions to separate lib, and reuse it in typestyle?

PS. This is a successful issue number =)

**Comment by whitecolor** [comment]
@basarat @notoriousb1t what is your opinion?

**Comment by bradleyayers** [comment]
I'm leaning towards "no" for this. Purely down to the increased maintenance of having to update two packages instead of one (especially given updates happens quite often).

If the cost of updating definitions stayed the same as it is today, then I'd have no objection.

In regards to snabbdom/snabbdom#339, option 2 of just "depending on typestyle but only using the types" sounds quite simple and pragmatic. That seems like a good starting point, and if problems are encountered along the way then we can address those case-by-case.

**Comment by whitecolor** [comment]
> If the cost of updating definitions stayed the same as it is today, then I'd have no objection.

Mostly if you need to update css props you will just have to update one package, update it version and publish. Typestyle package itself may depend on `~1.x.x` version so there will be no need update it until there is some breaking change in props definitions or something like that that will cause major version change. Though typestyle functionality itself almost doesn't dependepend at all on css props definitons.

> depending on typestyle but only using the types

This quite ok, but still requires some potentially unnecessary deps. And also there is special  `$unique` field in CSS props specific for `typestyle`.

**Comment by basarat** [comment]
My answer was / is no https://github.com/typestyle/typestyle/issues/37#issuecomment-262188166 

Leaving it open for searchability :heart:

I will probably only do it if I find a CSSinJS Solution other than TypeStyle that I really really really want to use :rose:

**Comment by notoriousb1t** [comment]
I don't have an issue with creating a separate repo within the TypeStyle org for the CSS Prop definitions.  In fact, this may be a good move so CSX doesn't have a direct dev dependency on TypeStyle.  I believe it is possible to do this and to extend those definitions in TypeStyle for nesting and special props.  

However, I don't think that moving this to an external repo will be good for the project.  Some of our API biases show up in the typed definitions, and putting our lots together in this manner is unlikely to be healthy for either project in the long term.

**Comment by whitecolor** [comment]
@notoriousb1t 
> I don't have an issue with creating a separate repo within the TypeStyle org for the CSS Prop definitions. In fact, this may be a good move so CSX doesn't have a direct dev dependency on TypeStyle. 

> However, I don't think that moving this to an external repo will be good for the project. 

I don't get, are those mutually exclusive paragraphs? )

**Comment by notoriousb1t** [comment]
Let me rephrase.  I do not have an issue with separating the definitions into a separate repo.  I **do** have an issue with maintaining that as part of different project.

**Comment by whitecolor** [comment]
Right ok, I think there is no reason to talk about *different project*. It is good to have definitions in a typestyle org and have it available as a separate package.

**Comment by whitecolor** [comment]
@notoriousb1t so what do you think is needed to undertake this step?

**Comment by pelotom** [comment]
FYI, this project looks very promising as an independent single source of truth that all CSS-in-JS libraries (as well as React) could use for their typings: https://github.com/frenic/csstype

**Comment by basarat** [comment]
If someone else creates an excellent type definition that receives consensus among CSS in JS projects we can definitely consider taking it as a dependency :rose:

**Comment by whitecolor** [comment]
@notoriousb1t 

So as I see typestyle from the recent time depends on `csstype`, which is great.
